### PR TITLE
Update Pipeline Migration Page

### PR DESCRIPTION
**Pull request by fa9r**
## Describe changes
I updated the pipeline migration page to include changes from #1653, #1648, and #1635 as well as a general overview comparison. I also turned all before-after comparisons into tabs.

Preliminary docs page: https://app.gitbook.com/o/-MCl1Hlw9oU4xibZ-ymf/s/vlf6HVLGPRknfyMwMtR2/user-guide/advanced-guide/migrate-your-old-pipelines-and-steps

## Pre-requisites
Please ensure you have done the following:
- [x] I have read the **CONTRIBUTING.md** document.
- [x] If my change requires a change to docs, I have updated the documentation accordingly.
- [ ] If I have added an integration, I have updated the [integrations](https://docs.zenml.io/component-gallery/integrations) table and the [corresponding website section](https://zenml.io/integrations).
- [ ] I have added tests to cover my changes.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Other (add details above)

### CODE REVIEW

**Review by schustmi — APPROVED** [review]

File: docs/book/user-guide/advanced-guide/migrate-your-old-pipelines-and-steps.md [hunk 1]
```diff
@@ -48,42 +133,53 @@ def my_step(param_1: int, param_2: Optional[float] = None) -> None:
...
+{% tabs %}
+{% tab title="Old Syntax" %}
 ```python
-# Old: The pipeline function gets steps as inputs and calls
-# the passed steps
 from zenml.pipelines import pipeline
 
 @pipeline
-def my_pipeline(my_step):
+def my_pipeline(my_step):  # Old: steps are argument of the pipeline function
```
Comment (bcdurak) [inline docs/book/user-guide/advanced-guide/migrate-your-old-pipelines-and-steps.md:177]:
```suggestion
def my_pipeline(my_step):  # Old: steps are arguments of the pipeline function
```

File: docs/book/user-guide/advanced-guide/migrate-your-old-pipelines-and-steps.md [hunk 2]
```diff
@@ -187,42 +305,157 @@ def my_step() -> None:
...
+{% tab title="New Syntax" %}
+```python
 from zenml import pipeline
 
 @pipeline
 def my_pipeline():
     step_1()
     step_2()
-    step_3(after=["step_1", "step_2"])
+    step_3(after=["step_1", "step_2"])  # New: Pass `after` argument when calling a step
```
Comment (bcdurak) [inline docs/book/user-guide/advanced-guide/migrate-your-old-pipelines-and-steps.md:380]:
```suggestion
    step_3(after=["step_1", "step_2"])  # New: Pass the `after` argument when calling a step
```

File: docs/book/user-guide/advanced-guide/migrate-your-old-pipelines-and-steps.md [hunk 3]
```diff
@@ -187,42 +305,157 @@ def my_step() -> None:
...
 my_pipeline = my_pipeline.with_options(schedule=schedule)
 my_pipeline()
 ```
+{% endtab %}
+{% endtabs %}
 
 Check out [this page](./schedule-pipeline-runs.md)
 for more information on how to schedule your pipelines.
 
+## Fetching Pipelines after Execution
```
Comment (bcdurak) [inline docs/book/user-guide/advanced-guide/migrate-your-old-pipelines-and-steps.md:319]:
```suggestion
## Fetching pipelines after execution
```

File: docs/book/user-guide/advanced-guide/migrate-your-old-pipelines-and-steps.md [hunk 4]
```diff
@@ -187,42 +305,157 @@ def my_step() -> None:
...
+]:
+    ...
+```
+{% endtab %}
+{% endtabs %}
+
+Check out [this page](./configure-steps-pipelines.md#type-annotations)
+for more information on how to annotate your step outputs.
+
+## Accessing Run information inside steps
```
Comment (bcdurak) [inline docs/book/user-guide/advanced-guide/migrate-your-old-pipelines-and-steps.md:427]:
```suggestion
## Accessing run information inside steps
```

**Review by bcdurak — APPROVED** [review]
Approved with a few small suggestions.

**Pull request merged**